packages/cozy-stack-client/src/SharingCollection.js

```diff
@@ -28,28 +28,54 @@ class SharingCollection extends DocumentCollection {
    * @param  {string} sharingType - If "two-way", will set the open_sharing attribute to true
    * @param  {string} description - Describes the sharing
    * @param  {string=} previewPath Relative URL of the sharings preview page
+   * @param  {boolean} open_sharing If the sharing is open or not
+   * @param  {boolean} forceOneWay By default, even if the sharing is `one-way`(aka read-only) we set the rules like
+   * a `two-way` (aka read/write) sharing since the stack is not able to update the rule later. By default, we still can
+   * make the sharing acting as a `one-way` for recipients by addind them to `read_only_recipients` attributes.
+   * If you really want to have a `one-way` sharing and you are sure you will not need to update it later, then you can use this
+   * `forceOneWay` attribue.
    */
   async share(
     document,
     recipients,
     sharingType,
     description,
-    previewPath = null
+    previewPath = null,
+    open_sharing,
+    forceOneWay = false
```

Comment:
I have trouble with this terminology "one-way", "two-way"; we use the read-only terminology elsewhere.

Furthermore, I think it is not very understandable to have:

```
share(
  document,
  recipients,
  'one-way',
  description,
   null,
   true,
   true
)
```

Even when using an option object, I think it is hard to understand what is the intention here:

```
share({
  document,
  recipients,
  sharingType: 'one-way',
  description,
  openSharing: true,
  forceOneWay: true
})
```

I am not sure but I think this would produce an invalid sharing ?

Reply:
> I am not sure but I think this would produce an invalid sharing ?

Nop. I think we can have an openSharing with a `one-way` sharingType and a forceOneWay